pkg/login/social/errors.go

```diff
@@ -2,29 +2,22 @@ package social
 
 import (
 	"errors"
-	"fmt"
+
+	"github.com/grafana/grafana/pkg/util/errutil"
 )
 
 var (
 	ErrIDTokenNotFound = errors.New("id_token not found")
 	ErrEmailNotFound   = errors.New("error getting user info: no email found in access token")
-)
 
-type InvalidBasicRoleError struct {
-	idP          string
-	assignedRole string
-}
+	errRoleAttributePathNotSet = errutil.NewBase(errutil.StatusBadRequest,
+		"oauth.role_attribute_path_not_set",
+		errutil.WithPublicMessage("Instance role_attribute_path misconfigured, please contact your administrator"))
```

Comment:
I'm not sure the errors will be parsed correctly, as we seem to use `errutil` error where this error is returned: https://github.com/grafana/grafana/blob/main/pkg/services/authn/clients/oauth.go#L125

Maybe it's fine, but worth testing (I can give it a go if you want).